connection.go

```diff
@@ -258,17 +257,23 @@ func NewTLSConnection(srvAddr string, rootCerts []byte,
 		connectNow, wef, log, tagsFunc)
 }
 
-// NewTLSConnectionWithServerName returns a connection that tries to connect to the
-// given server address with TLS. You can specify a server name that you expect
-// the certificate to contain.
-func NewTLSConnectionWithServerName(srvAddr string, srvName string, rootCerts []byte,
+func copyTLSConfig(c *tls.Config) *tls.Config {
```

Comment:
`tls.Config` contains a ton of reference types (maps, slices, etc.) so those could still be modified. So I don't know if doing the copying here really buys anything.